### PR TITLE
[API Compatibility] paddle.sigmoid sink into C++ 

### PR DESCRIPTION
**Pull request by hushenwei2000**
### PR Category
Operator Mechanism

### PR Types
New features

### Description
下沉paddle.sigmoid 到C++层，支持Torch的传参方法。

pcard-67164

### CODE REVIEW

**Review by wanghuancoder — APPROVED** [review]
LGTM

**Review by DanielSun11 — APPROVED** [review]
LGTM     等https://github.com/PaddlePaddle/Paddle/pull/74822 合入后需要补一个nn.functional.sigmoid的单测
